compiler/rustc_middle/src/ty/print/pretty.rs

```diff
@@ -887,26 +887,31 @@ pub trait PrettyPrinter<'tcx>: Printer<'tcx> + fmt::Write {
                 p!(write("{{"));
                 if !self.should_print_verbose() {
                     p!(write("closure"));
-                    // FIXME(eddyb) should use `def_span`.
-                    if let Some(did) = did.as_local() {
-                        if self.tcx().sess.opts.unstable_opts.span_free_formats {
-                            p!("@", print_def_path(did.to_def_id(), args));
-                        } else {
-                            let span = self.tcx().def_span(did);
-                            let preference = if with_forced_trimmed_paths() {
-                                FileNameDisplayPreference::Short
+                    if self.should_truncate() {
+                        write!(self, "@...}}")?;
+                        return Ok(());
+                    } else {
+                        // FIXME(eddyb) should use `def_span`.
```

Comment:
Question: is that FIXME still accurate, because I see a `def_span` invocation below?

Reply:
I noticed that too. I *don't know*. There's a call to `def_span`, but I'm unsure if it is also meant to apply to non-local types.

Reply:
I didn't bother digging earlier than 55871aad9a9c621e245c2b238f04f9fab55da759, but at *that* moment of time, we certainly weren't using `def_span`. Along the 6 years since that, this comment got shuffled around so I think it became outdated but no one noticed. We are now indeed properly printing the def path through `def_span`.